### PR TITLE
Closes #9014

### LINKED ISSUE #9014: `override` should not be allowed on static methods
```haxe
class Parent {}
class Main extends Parent {
	override static function main() { // no error here
		trace('hello');
	}
}
```
This should emit an error.

### PR DESCRIPTION
**Pull request by RblSb**

### CODE REVIEW

**Comment by Gama11** [comment]
Might be nice to have a test case in [`tests/misc/projects`](https://github.com/HaxeFoundation/haxe/tree/development/tests/misc/projects) (where we can check for expected compiler errors)?

File: src/typing/typeloadFields.ml [hunk 1]
```diff
@@ -1101,6 +1101,11 @@ let create_method (ctx,cctx,fctx) c f fd p =
 	let parent = (if not fctx.is_static then get_parent c (fst f.cff_name) else None) in
 	let dynamic = List.mem_assoc ADynamic f.cff_access || (match parent with Some { cf_kind = Method MethDynamic } -> true | _ -> false) in
 	if fctx.is_inline && dynamic then error (fst f.cff_name ^ ": You can't have both 'inline' and 'dynamic'") p;
+	let is_override = match fctx.override with
+		| None -> false
+		| _ -> true in
+	if (is_override && fctx.is_static) then error (fst f.cff_name ^ ": You can't have both 'override' and 'static'") p;
```
Comment (Aurel300) [inline src/typing/typeloadFields.ml]: This "You can't have both …" formulation seems a bit strange/informal. How about "`X` and `Y` cannot be applied to the same method" or something along those words?

Comment (Gama11) [reply src/typing/typeloadFields.ml]: That on the other hand might be a bit _too_ formal. :) Maybe something like:

> `override` is not allowed on `static` functions

Comment (Gama11) [reply src/typing/typeloadFields.ml]: Doing a quick Ctrl+Shift+F, "is not allowed on" seems to be a common pattern for Haxe error messages anyway.

File: src/typing/typeloadFields.ml [hunk 2]
```diff
@@ -1101,6 +1101,11 @@ let create_method (ctx,cctx,fctx) c f fd p =
 	let parent = (if not fctx.is_static then get_parent c (fst f.cff_name) else None) in
 	let dynamic = List.mem_assoc ADynamic f.cff_access || (match parent with Some { cf_kind = Method MethDynamic } -> true | _ -> false) in
 	if fctx.is_inline && dynamic then error (fst f.cff_name ^ ": You can't have both 'inline' and 'dynamic'") p;
+	let is_override = match fctx.override with
```
Comment (nadako) [inline src/typing/typeloadFields.ml]: `Option.is_some`? :)

**Review by Gama11 — APPROVED** [review]

**Pull request merged**